File: com.cordova.ble/www/ble.js
var argscheck = require('cordova/argscheck'),
utils = require('cordova/utils'),
exec = require('cordova/exec');

var Ble = function () {
};

var Tag = function (id, name, status){
    
    this.id = id || "";
    this.name = name || "";
    this.status = status || "";
};

var OBContent = function (title,uuid,message,path,action,createTime,contentState,contentType){
    
    
    this.title = title || "";
    this.uuid = uuid || "";
    this.message = message || "";
    this.path = path || "";
    this.action = action || "";
    this.createTime = createTime || "";
    this.contentState = contentState || "";
    this.contentType = contentType || "";
    
    
};



/**
 * @Platform iOS
 * startServiceWithClientID
 * @param SA_CLIENTID
 * @param SA_SECRET
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */

Ble.startServiceWithClientID = function (clientid, clientsecret, success, fail) {
    exec(success, fail, 'Ble', 'startServiceWithClientID', [clientid,clientsecret]);
};

/**
 * @Platform iOS
 * resetService
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */

Ble.resetService = function (success, fail) {
    exec(success, fail, 'Ble', 'resetService', []);
};


/**
 * @Platform iOS
 * viewControllerForContent
 * @param {OBContent} ob_content
 * @param {Function} successCallback
 * @param {Function} errorCallback
 * @return UIViewController
 */
Ble.viewControllerForContent = function (ob_content, success, fail) {
    exec(success, fail, 'Ble', 'viewControllerForContent', [ob_content]);
};

/**
 * @Platform iOS
 * getTags
 * @param {Function} successCallback
 * @param {Function} errorCallback
 * @return array of tags
 */
Ble.getTags = function ( success, fail) {
    exec(success, fail, 'Ble', 'getTags ', []);
};


/**
 * @Platform iOS
 * getSelectedTags
 * @param {Function} successCallback
 * @param {Function} errorCallback
 * @return array of selected tags
 */
Ble.getSelectedTags = function ( success, fail) {
    exec(success, fail, 'Ble', 'getSelectedTags ', []);
};

/**
 * @Platform iOS
 * setTags
 * @param {nsset} tags [[id,name],[id,name]]
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.setTags = function ( tags,success, fail) {
    exec(success, fail, 'Ble', 'setTags ', tags);
};

/**
 * @Platform iOS
 * clearCoupons
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.clearCoupons = function ( success, fail) {
    exec(success, fail, 'Ble', 'clearCoupons ', []);
};



/**
 * @Platform iOS
 * deleteContent
 * @param {OBContent} ob_content
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.deleteContent = function (ob_content, success, fail) {
    exec(success, fail, 'Ble', 'deleteContent', [ob_content]);
};


/**
 * @Platform iOS
 * contentOpened
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.contentOpened = function ( ob_content,success, fail) {
    exec(success, fail, 'Ble', 'contentOpened ', [ob_content]);
};

/**
 * @Platform iOS
 * contentTapped
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.contentTapped = function ( ob_content,success, fail) {
    exec(success, fail, 'Ble', 'contentTapped ', [ob_content]);
};

/**
 * @Platform iOS
 * showContentInfo
 * @param {OBContent} ob_content
 * @param {UIViewController} viewController
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.showContentInfo = function (ob_content, viewController, success, fail) {
    exec(success, fail, 'Ble', 'showContentInfo ', [ob_content,viewController]);
};

/**
 * @Platform iOS
 * getContent
 * @param {Function} successCallback
 * @param {Function} errorCallback
 * @return Content Array
 */
Ble.getContent = function (success, fail) {
    exec(success, fail, 'Ble', 'getContent', []);
};

/**
 * @Platform iOS
 * version
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.version = function ( success, fail) {
    exec(success, fail, 'Ble', 'version', []);
};

/**
 * @Platform iOS
 * sendUserMetrics
 * @param {NSDictionary} user [{.....}]
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.sendUserMetrics = function (user, success, fail) {
    exec(success, fail, 'Ble', 'sendUserMetrics', [user]);
};


/**
 * @Platform iOS
 * registerForPushNotificationWithDeviceToken
 * @param {NSString} token
 * @param {NSString} pushProvider
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.registerForPushNotificationWithDeviceToken = function (token, pushProvider, success, fail) {
    exec(success, fail, 'Ble', 'sendPushNotificationProviderDeviceToken', [token,pushProvider]);
};

/**
 * @Platform iOS
 * sendPushNotificationProviderDeviceToken
 * @param {NSDictionary} providerDeviceToken
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.sendPushNotificationProviderDeviceToken = function (token, success, fail) {
    exec(success, fail, 'Ble', 'sendPushNotificationProviderDeviceToken', [token]);
};

/**
 * @Platform iOS
 * sendReport
 * @param {NSString} data
 * @param {NSString} reporter
 * @param {NSString} message
 * @param {Function} errorCallback
 */
Ble.sendReport = function ( data,reporter,message, success, fail) {
    exec(success, fail, 'Ble', 'sendReport', [data,reporter,message]);
};

/**
 * @Platform iOS
 * viewControllerForTags
 * @param {Function} successCallback
 * @param {Function} errorCallback
 * @return UIViewController
 */
Ble.viewControllerForTags = function ( success, fail) {
    exec(success, fail, 'Ble', 'viewControllerForTags', []);
};


/**
 * @Platform iOS
 * requestAlwaysAuthorization
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.requestAlwaysAuthorization = function (success, fail) {
    exec(success, fail, 'Ble', 'requestAlwaysAuthorization', []);
};


/**
 * @Platform iOS
 * applicationDidEnterBackground
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.applicationDidEnterBackground = function ( success, fail) {
    exec(success, fail, 'Ble', 'applicationDidEnterBackground', []);
};

/**
 * @Platform iOS
 * applicationWillEnterForeground
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.applicationWillEnterForeground = function (success, fail) {
    exec(success, fail, 'Ble', 'applicationWillEnterForeground', []);
};

/**
 * @Platform iOS
 * setLogger
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.setLogger = function ( success, fail) {
    exec(success, fail, 'Ble', 'setLogger', []);
};

/**
 * @Platform iOS
 * requestWhenInUseAuthorization
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.requestWhenInUseAuthorization = function (success, fail) {
    exec(success, fail, 'Ble', 'requestWhenInUseAuthorization', []);
};

/**
 * initSDK
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.initSDK = function (success, fail) {
    exec(success, fail, 'Ble', 'initSDK', []);
};

/**
 * isBluetoothAvailable
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.isBluetoothAvailable = function (success, fail) {
    exec(success, fail, 'Ble', 'isBluetoothAvailable', []);
};

/**
 * startScan
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.startScan = function (success, fail) {
    exec(success, fail, 'Ble', 'startScan', []);
};

/**
 * stopScan
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.stopScan = function (success, fail) {
    exec(success, fail, 'Ble', 'stopScan', []);
};

/**
 * setBackgroundBetweenScanPeriod
 * @param {int} period
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.setBackgroundBetweenScanPeriod = function (period, success, fail) {
    exec(success, fail, 'Ble', 'setBackgroundBetweenScanPeriod', [period]);
};

/**
 * setForegroundMode
 * @param {Boolean} is
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.setForegroundMode = function (is, success, fail) {
    exec(success, fail, 'Ble', 'setForegroundMode', [is]);
};

/**
 * setCouponEnabled
 * @param {Boolean} is
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.setCouponEnabled = function (is, success, fail) {
    exec(success, fail, 'Ble', 'setCouponEnabled', [is]);
};

/**
 * enableGeofencing
 * @param {Boolean} is
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.enableGeofencing = function (is, success, fail) {
    exec(success, fail, 'Ble', 'enableGeofencing', [is]);
};
/**
 * setLocationTrackingEnabled
 * @param {Boolean} is
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.setLocationTrackingEnabled = function (is, success, fail) {
    exec(success, fail, 'Ble', 'setLocationTrackingEnabled', [is]);
};

/**
 * setAPIContentEnabled
 * @param {Boolean} is
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.setAPIContentEnabled = function (is, success, fail) {
    exec(success, fail, 'Ble', 'setAPIContentEnabled', [is]);
};


/**
 * enableBluetooth
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.enableBluetooth = function (success, fail) {
    exec(success, fail, 'Ble', 'enableBluetooth', []);
};


/**
 * getTags
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.getTags = function (success, fail) {
    exec(success, fail, 'Ble', 'getTags', []);
};


/**
 * restartLocationTracking
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.restartLocationTracking = function (success, fail) {
    exec(success, fail, 'Ble', 'restartLocationTracking', []);
};

/**
 * logOut
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.logOut = function (success, fail) {
    exec(success, fail, 'Ble', 'logOut', []);
};

/**
 * sendGenericUserProfile
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.sendGenericUserProfile = function (profile,success, fail) {
    exec(success, fail, 'Ble', 'sendGenericUserProfile', [profile]);
};

/**
 * isInForeground
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.isInForeground = function (success, fail) {
    exec(success, fail, 'Ble', 'isInForeground', []);
};


/**
 * isCouponEnabled
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.isCouponEnabled = function (success, fail) {
    exec(success, fail, 'Ble', 'isCouponEnabled', []);
};

/**
 * isAPIEnabled
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.isAPIEnabled = function (success, fail) {
    exec(success, fail, 'Ble', 'isAPIEnabled', []);
};

/**
 * deleteCoupon
 * @param {Function} var1
 * @param {Function} var2
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.deleteCoupon = function (var1,var2,success, fail) {
    exec(success, fail, 'Ble', 'deleteCoupon', [var1,var2]);
};

/**
 * sendDeviceToken
 * @param {Function} var1
 * @param {Function} var2
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.sendDeviceToken = function (var1,var2,success, fail) {
    exec(success, fail, 'Ble', 'sendDeviceToken', [var1,var2]);
};

/**
 * setAuthExtraData
 * @param {Function} var1
 * @param {Function} var2
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.setAuthExtraData = function (var1,success, fail) {
    exec(success, fail, 'Ble', 'setAuthExtraData', [var1]);
};

/**
 * setAuthExtraData
 * @param {Function} var1, this is JSONARRAY of Tag [[id,name,state],[id,name,state],[id,name,state]]
 * @param {Function} var2
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.setTagsFilterForCoupons = function (tagArrayList,success, fail) {
    exec(success, fail, 'Ble', 'setAuthExtraData', [tagArrayList]);
};

/**
 * markAsTapped
 * @param {Function} var1
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.markAsTapped = function (var1,success, fail) {
    exec(success, fail, 'Ble', 'markAsTapped', [var1]);
};

/**
 * markAsOpened
 * @param {Function} var1
 * @param {Function} successCallback
 * @param {Function} errorCallback
 */
Ble.markAsOpened = function (var1,success, fail) {
    exec(success, fail, 'Ble', 'markAsOpened', [var1]);
};



/**
 * @Platform iOS
 * didRangeBeacons
 */

Ble.didRangeBeacons = function (beaconArray,ob_region) {
};

/**
 * @Platform iOS
 * locationManagerDidEnterRegion
 */
Ble.locationManagerDidEnterRegion = function (cl_region) {
};


/**
 * @Platform iOS
 * locationManagerDidEnterRegion
 */
Ble.locationManagerDidExitRegion = function (cl_region) {
};


/**
 * @Platform iOS
 * didReceiveContent
 */
Ble.didReceiveContent = function (couponsArray) {
    
};



/**
 * @Platform iOS
 * didRequestInfo
 */
Ble.didRequestInfo = function (ob_content,viewcontroller) {
    
};

/**
 * @Platform iOS
 * onTagsReceived
 */
Ble.onTagsReceived = function (tags) {
    
};


/**
 * @Platform iOS
 * onCouponsReceived
 */
Ble.onCouponsReceived = function (coupan, beacons) {
    
};



/**
 * @Platform iOS
 * onCouponsReceived
 */
Ble.onBluemixCredentialsReceived = function (blueMix) {
    
};

/**
 * @Platform iOS
 * didRangeBeaconsInRegion
 */
Ble.didRangeBeaconsInRegion = function (beacons) {
    
};

/**
 * @Platform iOS
 * onyxBeaconError
 */
Ble.onyxBeaconError = function (msg) {

};
/*
if (cordova.platformId === 'android' || cordova.platformId === 'amazon-fireos' || cordova.platformId === 'windowsphone') {
    
    var channel = require('cordova/channel');
    
    channel.createSticky('onBlePluginReady');
    channel.waitForInitialization('onBlePluginReady');
    
    channel.onCordovaReady.subscribe(function() {
                                     exec(Ble.didRangeBeacons, undefined, 'Ble', 'messageChannel', []);
                                     exec(Ble.locationManagerDidEnterRegion, undefined, 'Ble', 'messageChannel', []);
                                     exec(Ble.locationManagerDidExitRegion, undefined, 'Ble', 'messageChannel', []);
                                     exec(Ble.didReceiveContent, undefined, 'Ble', 'messageChannel', []);
                                     exec(Ble.didRequestInfo, undefined, 'Ble', 'messageChannel', []);
                                     exec(Ble.onyxBeaconError, undefined, 'Ble', 'messageChannel', []);
                                     channel.initializationComplete('onBlePluginReady');
                                     });
}
*/
module.exports = Ble;

File: com.cordova.ble/src/ios/Ble.m
//
//  Ble.m
//  onyxB
//
//  Created by Nomi on 21/11/2015.
//
//

#import "Ble.h"



@implementation Ble

/*
 NSDictionary *jsonObj = [ [NSDictionary alloc]
 initWithObjectsAndKeys :
 dateStr, @"dateStr",
 @"true", @"success",
 nil
 ];
 */

#pragma mark - Coupan View

- (void)getContent:(CDVInvokedUrlCommand *)command{
    
    NSArray *coupons = [[OnyxBeacon sharedInstance] getContent];
    
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsArray: coupons];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}



#pragma mark - OnyxBeaconCouponDelegate Methods
- (void)didRangeBeacons:(NSArray *)beacons inRegion:(OBBeaconRegion *)region {
    
    NSString* jsString = nil;
    jsString = [NSString stringWithFormat:@"%@(\"%@,%@\");", @"window.cordova.plugins.Ble.didRangeBeacons", beacons,region];
    [self.commandDelegate evalJs:jsString];
    
}


- (void)locationManagerDidEnterRegion:(CLRegion *)region{

    NSString* jsString = nil;
    jsString = [NSString stringWithFormat:@"%@(\"%@\");", @"window.cordova.plugins.Ble.locationManagerDidEnterRegion", region];
    [self.commandDelegate evalJs:jsString];
    
}




- (void)locationManagerDidExitRegion:(CLRegion *)region{
    
    NSString* jsString = nil;
    jsString = [NSString stringWithFormat:@"%@(\"%@\");", @"window.cordova.plugins.Ble.locationManagerDidExitRegion", region];
    [self.commandDelegate evalJs:jsString];
    
}





- (void)didReceiveContent:(NSArray *)coupons {
    /*
     for (OBContent *coupon in coupons) {
     UILocalNotification *notification = [[UILocalNotification alloc] init];
     notification.alertBody = coupon.message;
     notification.userInfo = @{@"uuid": coupon.uuid};
     notification.soundName = UILocalNotificationDefaultSoundName;
     [[UIApplication sharedApplication] presentLocalNotificationNow:notification];
     }
     */
    //Send Coupan to Cordova
    
    NSString* jsString = nil;
    jsString = [NSString stringWithFormat:@"%@(\"%@\");", @"window.cordova.plugins.Ble.didReceiveContent", coupons];
    [self.commandDelegate evalJs:jsString];
}

- (void)didRequestInfo:(OBContent *)content inViewController:(UIViewController *)viewController {
    NSString* jsString = nil;
    jsString = [NSString stringWithFormat:@"%@(\"%@,%@\");", @"window.cordova.plugins.Ble.didRequestInfo", content,viewController.restorationIdentifier];
    [self.commandDelegate evalJs:jsString];
}

- (void)contentOpened:(CDVInvokedUrlCommand *)command {
    OBContent *content = [OBContent alloc];

    
    
    [[OnyxBeacon sharedInstance] contentOpened:content];
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"contentOpened Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

- (void)getTags:(CDVInvokedUrlCommand *)command {
    NSArray * result  = [[OnyxBeacon sharedInstance] getTags];
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsArray:result];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}


- (void)getSelectedTags:(CDVInvokedUrlCommand *)command {
    NSArray * result  = [[[OnyxBeacon sharedInstance] getSelectedTags] allObjects];
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsArray:result];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

- (void)clearCoupons:(CDVInvokedUrlCommand *)command {
    [[OnyxBeacon sharedInstance] clearCoupons];
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"clearCoupons Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}



- (void)setTags:(CDVInvokedUrlCommand *)command {
    
    NSSet *set = [[NSSet alloc] init];
    [set setByAddingObjectsFromArray:command.arguments];
    [[OnyxBeacon sharedInstance] setTags:set];
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"setTags Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}
- (void)contentTapped:(CDVInvokedUrlCommand *)command {
    OBContent *content = [OBContent alloc];
    [[OnyxBeacon sharedInstance] contentTapped:content];
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"contentTapped Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

- (void)showContentInfo:(CDVInvokedUrlCommand *)command {
    OBContent *content = [OBContent alloc];
    UIWindow *window = [UIApplication sharedApplication].keyWindow;
    UIViewController *rootViewController = window.rootViewController;

    [[OnyxBeacon sharedInstance] showContentInfo:content inViewController:rootViewController ];
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"showContentInfo Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}


- (void)viewControllerForContent:(CDVInvokedUrlCommand *)command {
    OBContent *content = [OBContent alloc];
    UIViewController *c = [[OnyxBeacon sharedInstance] viewControllerForContent:content];
    
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: c.restorationIdentifier];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];

}

- (void)deleteContent:(CDVInvokedUrlCommand *)command {
    OBContent *content = [OBContent alloc];
    [[OnyxBeacon sharedInstance] deleteContent:content];
    
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"deleteContent Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}


#pragma mark - Main OnyxBeacon

- (void) version: (CDVInvokedUrlCommand *)command {
    
    [[OnyxBeacon sharedInstance] version];
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"Version Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
    
}

- (void)sendUserMetrics : (CDVInvokedUrlCommand *)command {
    NSMutableDictionary* user = [command.arguments objectAtIndex:0];

    [[OnyxBeacon sharedInstance] sendUserMetrics:user];
    
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"sendUserMetrics Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
    
}


- (void)registerForPushNotificationWithDeviceToken : (CDVInvokedUrlCommand *)command {
    NSData* token = [[command.arguments objectAtIndex:0] dataUsingEncoding:NSUTF8StringEncoding];
    NSString* provider = [command.arguments objectAtIndex:0];
    
    [[OnyxBeacon sharedInstance] registerForPushNotificationWithDeviceToken:token forProvider:provider handler:^(NSDictionary *d, NSError *e) {
        
    }];
    
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"registerForPushNotificationWithDeviceToken Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
    
}





- (void)sendPushNotificationProviderDeviceToken : (CDVInvokedUrlCommand *)command {
    NSString* token = [command.arguments objectAtIndex:0] ;
    
    [[OnyxBeacon sharedInstance] sendPushNotificationProviderDeviceToken:token];
    
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"sendPushNotificationProviderDeviceToken Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
    
}


- (void)sendReport :(CDVInvokedUrlCommand *)command {
    
    NSData* data = [[command.arguments objectAtIndex:0] dataUsingEncoding:NSUTF8StringEncoding];
    NSString* reporter = [command.arguments objectAtIndex:1];
    NSString* message = [command.arguments objectAtIndex:2];
    
    [[OnyxBeacon sharedInstance] sendReport:data reporter:reporter message:message handler:^(NSError *error) {

    }];
    
    
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"sendLogReport Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}


- (void)viewControllerForTags:(CDVInvokedUrlCommand *)command
{
    UIViewController *c =  [[OnyxBeacon sharedInstance] viewControllerForTags];
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: c.restorationIdentifier];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];

    
}

- (void)requestWhenInUseAuthorization:(CDVInvokedUrlCommand *)command
{
    return [[OnyxBeacon sharedInstance] requestWhenInUseAuthorization];
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"requestWhenInUseAuthorization Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
    
}



- (void)onyxBeaconError:(NSError *)error {
    NSString* jsString = nil;
    jsString = [NSString stringWithFormat:@"%@(\"%@\");", @"window.cordova.plugins.Ble.onyxBeaconError", error];
    [self.commandDelegate evalJs:jsString];
    
}



-(void) requestAlwaysAuthorization :(CDVInvokedUrlCommand *)command {
    [[OnyxBeacon sharedInstance] requestAlwaysAuthorization];
    
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"requestAlwaysAuthorization Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

-(void) resetService:(CDVInvokedUrlCommand *)command {
    
    [[OnyxBeacon sharedInstance] resetService];
    
    
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"resetService Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
    
}

-(void) setLogger:(CDVInvokedUrlCommand *)command {
    
    [[OnyxBeacon sharedInstance] setLogger:^(NSString *message) {
        
    }];
    
    
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"setLogger Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
    
}

-(void) startServiceWithClientID:(CDVInvokedUrlCommand *)command {
    
    NSString* SA_CLIENTID = [command.arguments objectAtIndex:0];
    NSString* SA_SECRET = [command.arguments objectAtIndex:1];
    
    
    [[OnyxBeacon sharedInstance] startServiceWithClientID:SA_CLIENTID secret:SA_SECRET];
    [[OnyxBeacon sharedInstance] setContentDelegate:self];
    [[OnyxBeacon sharedInstance] setDelegate:self];
    
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"startServiceWithClientID Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
    
}


- (void)applicationDidEnterBackground:(CDVInvokedUrlCommand *)command
{
    [[OnyxBeacon sharedInstance] didEnterBackground];
    
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"didEnterBackground Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}


- (void)applicationWillEnterForeground:(CDVInvokedUrlCommand *)command
{
    [[OnyxBeacon sharedInstance] willEnterForeground];
    
    CDVPluginResult* pluginResult = [CDVPluginResult
                                     resultWithStatus:CDVCommandStatus_OK
                                     messageAsString: @"didEnterBackground Invoked"];
    
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
    
}

@end